### PR TITLE
(feature) O3-2965:Appointments: Have a way to disable Check In and Ch…

### PR DESCRIPTION
**Pull request by mogoodrich**
…eck Out buttons

See: https://openmrs.atlassian.net/browse/O3-2965

Note that is PR is based off of O3-2964, so I will merge these in together when merging, (or handle any rebasing if necessary)

## Requirements

- [ ] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [ ] My work includes tests or is validated by existing tests.

### CODE REVIEW

File: packages/esm-appointments-app/src/config-schema.ts [hunk 1]
```diff
@@ -90,6 +90,30 @@ export const configSchema = {
     _description: 'Whether to allow scheduling of all-day appointments (vs appointments with start time and end time)',
     _default: true,
   },
+  checkInButton: {
+    enabled: {
+      _type: Type.Boolean,
+      _description: 'Whether the check-in button on the "Appointments" list should be enabled',
+      _default: true,
+    },
+    customUrl: {
```
Comment (mogoodrich) [inline packages/esm-appointments-app/src/config-schema.ts:99]: I haven't implemented this custom URL functionality yet, but hope to have a separate ticket and PR out for that today or tomorrow... if we decided we don't want/can't do this, i will remove this parameter and the one below.

**Review by mseaton — APPROVED** [review]
LGTM, thanks @mogoodrich

File: packages/esm-appointments-app/src/appointments/common-components/appointments-actions.component.tsx [hunk 2]
```diff
@@ -22,6 +22,7 @@ interface AppointmentActionsProps {
 
 const AppointmentActions: React.FC<AppointmentActionsProps> = ({ appointment }) => {
   const { t } = useTranslation();
+  const { checkInButton, checkOutButton } = useConfig();
```
Comment (chibongho) [inline packages/esm-appointments-app/src/appointments/common-components/appointments-actions.component.tsx:25]: minor: type the `useConfig` whenever possible: `useConfig<ConfigObject>()`, (and make sure to import the right `ConfigObject` from appointments)

File: packages/esm-appointments-app/src/appointments/common-components/appointments-actions.test.tsx [hunk 3]
```diff
@@ -66,21 +73,43 @@ describe('AppointmentActions', () => {
...
+      visits: [],
+    }));
+    const props = { ...defaultProps };
+    const { queryByText } = render(<AppointmentActions {...props} />);
+    const button = queryByText('Check In');
+    expect(button).not.toBeInTheDocument();
   });
 
-  it('renders the correct button when the patient has checked out', () => {
+  it('renders the checked out button when the patient has checked out', () => {
```
Comment (chibongho) [inline packages/esm-appointments-app/src/appointments/common-components/appointments-actions.test.tsx:108]: shouldn't be a button at this point, right?

**Review by chibongho — APPROVED** [review]
LGTM. minor comments.

File: packages/esm-appointments-app/src/appointments/common-components/appointments-actions.test.tsx (hunk 3 above)
Comment (mogoodrich) [reply packages/esm-appointments-app/src/appointments/common-components/appointments-actions.test.tsx:108]: One might thing that, but it's a non-clickable button, maybe for styling reasons?

File: packages/esm-appointments-app/src/appointments/common-components/appointments-actions.component.tsx (hunk 2 above)
Comment (mogoodrich) [reply packages/esm-appointments-app/src/appointments/common-components/appointments-actions.component.tsx:25]: I think you mentioned this before, will fix.

Comment (mogoodrich) [reply packages/esm-appointments-app/src/appointments/common-components/appointments-actions.component.tsx:25]: Handling this in the follow-on PR for O3-2966

**Comment by mogoodrich** [comment]
Thanks @chibongho  and @mseaton , merging!  @ibacher  @denniskigen  post-commit reviews still welcome...

**Pull request merged**